### DISCUSSION TITLE
Commits with stacktraces (which match `#\(0-9\*\)`) have spammed the issues pages with tons of false `...added a commit that referenced this issue on...`s, what to do?

### DISCUSSION THREAD

**Discussion by SwuduSusuwu** [community/community#143711] [New to GitHub]
https://github.com/SwuduSusuwu/SubStack/issues/3#ref-commit-f57e789 (from months ago) has 6 such in a row. Have not found options to remove those.

Just the last message (at the bottom, from a commit just now) has to do with the issue.

**Comment by lucatosc** [comment]
It sounds like you're dealing with a situation where commit messages are auto-generating issue references, potentially leading to spam or clutter in your issue tracker. Here are some steps you can take to address this problem:

1. **Identify the Source of the Commits**:
   - Identify which repository or CI/CD pipeline is generating these commits. This might involve checking your project's recent commits or looking into recent builds if they are automated.

2. **Update Commit Message Guidelines**:
   - If the commits are coming from team members, have a discussion about the importance of concise and relevant commit messages. Establish a standard for commit messages to prevent them from unintentionally referencing issues.

3. **Modify Automation Scripts**:
   - If the commits are being created by automated scripts (like CI/CD jobs), review and adjust the logic that generates commit messages. Ensure that it does not include unintended issue references or follow a format that leads to false positives.

4. **Adjust Issue Tracking Settings**:
   - Some issue tracking systems might have settings to filter or ignore certain references in commit messages. Look into the documentation for your issue tracking tool (like GitHub, GitLab, JIRA, etc.) for settings on how to manage commit message references.

5. **Cleanup Existing Spam Issues**:
   - Open the issue tracker and identify the issues that have been cluttered with these spam comments. Depending on the volume, you might consider:
     - Manually closing or deleting the irrelevant comments.
     - Using bulk actions provided by the issue tracking system to clear out or manage these entries.

6. **Create a Filter or Query**:
   - If applicable, create a query/filter in your issue tracker to help identify issues affected by this spam, so you can keep it organized and easier to manage in the future.

7. **Communicate with Your Team**:
   - Ensure your team is aware of the changes. Share the guidelines or updated practices, and encourage them to follow them to avoid this issue in the future.

8. **Monitor for Future Spam**:
   - Keep an eye on future commits and issue tracker entries to ensure the problem does not recur. You may want to set up alerts for frequent references or unwanted patterns.

By following these steps, you should be able to minimize the spam caused by irrelevant commit references and enhance your project's issue management process.

**Reply by SwuduSusuwu** [reply]
> It sounds like you're dealing with a situation where commit messages are auto-generating issue references, potentially leading to spam or clutter in your issue tracker. Here are some steps you can take to address this problem:
> 
>     1. **Identify the Source of the Commits**:
>        
>        * Identify which repository or CI/CD pipeline is generating these commits. This might involve checking your project's recent commits or looking into recent builds if they are automated.

![Screenshot_20241105-200711](https://github.com/user-attachments/assets/8799f231-292f-471e-ba80-de759bae561b)


It is from `clang++` fsan, such as:
```
=================================================================
==18709==ERROR: AddressSanitizer: stack-use-after-scope on address 0x007ffc3d9511 at pc 0x007450ea2a78 bp 0x007ffc3d7e90 sp 0x007ffc3d7678
READ of size 1 at 0x007ffc3d9511 thread T0
    #0 0x7450ea2a74 in strncmp out/lib/compiler-rt-aarch64/out/llvm-project/compiler-rt/lib/asan/../sanitizer_common/sanitizer_common_interceptors.inc:545:3
    https://github.com/SwuduSusuwu/SubStack/issues/1 0x745388e200  (/data/data/com.termux/files/usr/lib/libtermux-exec.so+0x2200)
    https://github.com/SwuduSusuwu/SubStack/issues/2 0x745388dd68 in execve (/data/data/com.termux/files/usr/lib/libtermux-exec.so+0x1d68)
    https://github.com/SwuduSusuwu/SubStack/issues/3 0x63f79ca63c in Susuwu::execves(std::__ndk1::vector<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const, std::__ndk1::allocator<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const>> const&, std::__ndk1::vector<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const, std::__ndk1::allocator<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const>> const&) /data/data/com.termux/files/home/SubStack/./cxx/ClassCns.cxx:34:2
    https://github.com/SwuduSusuwu/SubStack/issues/4 0x63f7a3a7ac in Susuwu::testHarnesses() /data/data/com.termux/files/home/SubStack/./cxx/main.cxx:21:7
    https://github.com/SwuduSusuwu/SubStack/issues/5 0x63f7a3b1e0 in main /data/data/com.termux/files/home/SubStack/./cxx/main.cxx:40:9
    https://github.com/SwuduSusuwu/SubStack/issues/6 0x74524c9e18 in __libc_init (/apex/com.android.runtime/lib64/bionic/libc.so+0x56e18) (BuildId: 33ad5959e2b38fc822cda3c642e16c94)
```
> 
>     2. **Update Commit Message Guidelines**:
>        
>        * If the commits are coming from team members, have a discussion about the importance of concise and relevant commit messages. Establish a standard for commit messages to prevent them from unintentionally referencing issues.
> 

The new standard is to escape (put spaces after `#`) the bits of the stacktraces which match the regex.

>     3.  ... If the commits are being created by automated scripts

The messages come from _GitHub_'s bot (which uses regexes,) but commits = manual.
>     4. **Adjust Issue Tracking Settings**:
>        
>        * Some issue tracking systems might have settings to filter or ignore certain references in commit messages. Look into the documentation for your issue tracking tool (like GitHub, GitLab, JIRA, etc.) for settings on how to manage commit message references.

All of the results were irrelevent when searched how to do this.
>     5. **Cleanup Existing Spam Issues**:
>        
>        * Open the issue tracker and identify the issues that have been cluttered with these spam comments. Depending on the volume, you might consider:
>          
>          * Manually closing or deleting the irrelevant comments.
>          * Using bulk actions provided by the issue tracking system to clear out or manage these entries.

Wish to do this, but those autonomous messages have no options to report or delete.

>     6. **Create a Filter or Query**:
>        
>        * If applicable, create a query/filter in your issue tracker to help identify issues affected by this spam, so you can keep it organized and easier to manage in the future.

Will do.
>     7. **Communicate with Your Team**:
>        
>        * Ensure your team is aware of the changes. Share the guidelines or updated practices, and encourage them to follow them to avoid this issue in the future.
> 

Am solo.
>     8. **Monitor for Future Spam**:
>        
>        ... You may want to set up alerts for frequent references or unwanted patterns.

Have not found how to.
